internal/store/atlas/store.go

```diff
@@ -0,0 +1,498 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atlas
+
+//go:generate mockgen -destination=../../mocks/atlas/store.go -package=atlas github.com/mongodb/mongodb-atlas-cli/internal/store/atlas CredentialsGetter
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/mongodb-forks/digest"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	atlasauth "go.mongodb.org/atlas/auth"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
+	"go.mongodb.org/ops-manager/opsmngr"
+)
+
+const (
+	yes                       = "yes"
+	responseHeaderTimeout     = 1 * time.Minute
+	telemetryTimeout          = 1 * time.Second
+	tlsHandshakeTimeout       = 5 * time.Second
+	timeout                   = 5 * time.Second
+	keepAlive                 = 30 * time.Second
+	maxIdleConns              = 5
+	maxIdleConnsPerHost       = 4
+	idleConnTimeout           = 30 * time.Second
+	expectContinueTimeout     = 1 * time.Second
+	versionManifestStaticPath = "https://opsmanager.mongodb.com/"
```

Comment:
not needed for atlas

Reply:
Yes. I have attempted to clean up the ops manager but found out that there is also common logic used in atlas cli.
Logic uses opsManagerURL for atlas in the config that allows us to set custom url.